demos/sample_azure_iot_pnp/sample_azure_iot_pnp_simulated_data.c

```diff
@@ -0,0 +1,497 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+#include "sample_azure_iot_pnp_data_if.h"
+
+/* Standard includes. */
+#include <string.h>
+#include <stdio.h>
+
+/* Azure JSON includes */
+#include "azure_iot_json_reader.h"
+#include "azure_iot_json_writer.h"
+
+/* FreeRTOS */
+/* This task provides taskDISABLE_INTERRUPTS, used by configASSERT */
+#include "task.h"
```

Comment:
lets depend upon FreeRTOS.h that provides configASSERT

Reply:
I'll try that. If it works sure I'll replace.

Reply:
Seems fine.

Reply:
But btw task.h must continue there, otherwise the build will fail....

```
 /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/bin/ld: demos/projects/ST/b-l475e-iot01a/CMakeFiles/iot-middleware-sample-pnp.dir/__/__/__/sample_azure_iot_pnp/sample_azure_iot_pnp_simulated_data.c.obj: in function `prvSkipPropertyAndValue':
  sample_azure_iot_pnp_simulated_data.c:(.text.prvSkipPropertyAndValue+0x24): undefined reference to `taskDISABLE_INTERRUPTS'
  /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/bin/ld: sample_azure_iot_pnp_simulated_data.c:(.text.prvSkipPropertyAndValue+0x46): undefined reference to `taskDISABLE_INTERRUPTS'
  /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/bin/ld: sample_azure_iot_pnp_simulated_data.c:(.text.prvSkipPropertyAndValue+0x68): undefined reference to `taskDISABLE_INTERRUPTS'
  /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/bin/ld: demos/projects/ST/b-l475e-iot01a/CMakeFiles/iot-middleware-sample-pnp.dir/__/__/__/sample_azure_iot_pnp/sample_azure_iot_pnp_simulated_data.c.obj: in function `prvGetNewMaxTemp':
  sample_azure_iot_pnp_simulated_data.c:(.text.prvGetNewMaxTemp+0x36): undefined reference to `taskDISABLE_INTERRUPTS'
  /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/bin/ld: sample_azure_iot_pnp_simulated_data.c:(.text.prvGetNewMaxTemp+0x60): undefined reference to `taskDISABLE_INTERRUPTS'
  /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/bin/ld: demos/projects/ST/b-l475e-iot01a/CMakeFiles/iot-middleware-sample-pnp.dir/__/__/__/sample_azure_iot_pnp/sample_azure_iot_pnp_simulated_data.c.obj:sample_azure_iot_pnp_simulated_data.c:(.text.prvGetNewMaxTemp+0x8e): more undefined references to `taskDISABLE_INTERRUPTS' follow
  collect2: error: ld returned 1 exit status
```